### ISSUE TITLE
Sublime Text style auto completion for words and member functions with fuzzy matching

### ISSUE THREAD

**Issue by jasonhuhx** [Microsoft/vscode#22153]
- VSCode Version: 1.10.1
- OS Version: macOS

Is it possible for VSCode to support the following kinds of auto complete, please see screenshots

Word completion:

<img width="357" alt="screen shot 2017-03-07 at 10 23 19 am" src="https://cloud.githubusercontent.com/assets/1328495/23663299/30119be4-0320-11e7-8c11-5f4cd201dca8.png">

Member function completion:

<img width="359" alt="screen shot 2017-03-07 at 10 20 22 am" src="https://cloud.githubusercontent.com/assets/1328495/23663321/3d55fa2a-0320-11e7-8ad1-6449accbdb2e.png">

All auto-completions come with fuzzy matching.

Steps to Reproduce:

1.
2.

**Comment by csalv5** [comment]
Agree, actually fuzzy matching only works with camel Case, but in sublime and atom you can get completion for every middle letters.

I still miss this feature in vscode

**Comment by jrieken** [comment]
I like, tho will be some work and maybe some adjustment for folks that are used to our current filtering and ranking

**Comment by jrieken** [comment]
This got implemented with https://github.com/Microsoft/vscode/pull/22884

<img width="280" alt="screen shot 2017-03-20 at 17 38 54" src="https://cloud.githubusercontent.com/assets/1794099/24110615/2e491c96-0d94-11e7-939f-52cd9d5d76ab.png">

**Comment by csholmq** [comment]
@jrieken And this is valid for all languages then?

**Comment by jrieken** [comment]
Yes, we have changed the default matcher to be more fuzzy.

**Comment by schmuli** [comment]
@jrieken 

VSCode:
1.11.0-insider
6b4e8566986d1403d85859fbc383288acad480df

I appear to be getting stuck in a recursive loop, and according to DevTools (I clicked Pause in the Debugger) it is happening in [`_matchRecursive`](https://github.com/Microsoft/vscode/blob/master/src/vs/base/common/filters.ts#L425).

I tested this with `code-insiders --verbose --disable-extensions`.

I was able to reproduce this with a simple HTML file:

```html
<div>
    <form>
        <div>
            <button>Test</button>
            <ul>
                <li>
                    <a
                </li>
            </ul>
        </div>
    </form>
</div>
```

After typing the opening `<a` and IntelliSense appears, trying to type an additional character will freeze the whole VSCode.
